CalibTracker/SiPixelConnectivity/src/PixelToLNKAssociateFromAscii.cc

```diff
@@ -213,8 +215,10 @@ void PixelToLNKAssociateFromAscii::addConnections(
        }
        theConnection.push_back( make_pair(detectorRocId,cablingRocId));
      } 
+     if (!loopExecuted) delete name;
```

Comment:
The problem is here: https://github.com/cms-sw/cmssw/blob/master/CalibTracker/SiPixelConnectivity/src/PixelToLNKAssociateFromAscii.cc#L197

Of course I could replace it with a name.get() then, but I'm very afraid of that. When the unique_ptr goes out of scope, the memory that is still needed somewhere could be deallocated. Of course it's maybe possible to investigate case-by-case if that really happens, but it's not really easy to test locally so I was conservative with the unique pointers.

Reply:
Well that would mean copying a unique pointer, right? I will try later, but I think that is not the solution. Since I guess the idea of the pointer in the struct is that it has a small footprint when copying, I'm really not sure about that...

Reply:
No, that would mean moving the `unique_ptr`. But ok, it won't really work either, since the loop may execute multiple times, so for that route one would need `std::shared_ptr`, which is then rather far from being light-weight. So maybe any real improvement here would mean a more complete rethinking of the memory ownership model.

Reply:
Exactly! Rethinking the memory ownership model would be healthy in many parts of CMSSW actually :) But maybe not in this PR.